pkg/pillar/cmd/downloader/download.go

```diff
@@ -55,35 +55,32 @@ func download(ctx *downloaderContext, trType zedUpload.SyncTransportType,
 	}
 
 	req.Post()
-	for {
-		select {
-		case resp, ok := <-respChan:
-			if resp.IsDnUpdate() {
-				asize, osize, progress := resp.Progress()
-				log.Infof("Update progress for %v: %v/%v",
-					resp.GetLocalName(), asize, osize)
-				status.Progress(progress)
-				continue
-			}
-			if !ok {
-				errStr := fmt.Sprintf("respChan EOF for <%s>, <%s>, <%s>",
-					dpath, region, filename)
-				log.Errorln(errStr)
-				return errors.New(errStr)
-			}
-			if syncOp == zedUpload.SyncOpDownload {
-				err = resp.GetDnStatus()
-			} else {
-				_, err = resp.GetUpStatus()
-			}
-			if resp.IsError() {
-				return err
-			}
-			log.Infof("Done for %v: size %v/%v",
-				resp.GetLocalName(),
-				resp.GetAsize(), resp.GetOsize())
-			status.Progress(100)
-			return nil
+	for resp := range respChan {
+		if resp.IsDnUpdate() {
+			asize, osize, progress := resp.Progress()
+			log.Infof("Update progress for %v: %v/%v",
+				resp.GetLocalName(), asize, osize)
+			status.Progress(progress)
+			continue
 		}
+		if syncOp == zedUpload.SyncOpDownload {
+			err = resp.GetDnStatus()
+		} else {
+			_, err = resp.GetUpStatus()
+		}
+		if resp.IsError() {
+			return err
```

Comment:
print a log.Error here..

Reply:
I stuck with what we agreed on the supersized PR: distinct changes. This one covers only replacing the loop and any logic necessary to make that happen. We should open a separate PR to add just that.